agent_store/src/mongodb.rs

```diff
@@ -28,10 +28,24 @@ where
     }
 }
 
-pub struct MongoDB;
+#[derive(Clone)]
+pub struct MongoDB {
+    pub client: Client,
+}
+
+impl MongoDB {
+    pub async fn new() -> Self {
+        let connection_string = config().event_store.connection_string.clone().expect(
+            "Missing config parameter `event_store.connection_string` or `UNICORE__EVENT_STORE__CONNECTION_STRING`",
+        );
+        let client = default_mongo_client(&connection_string).await;
+        Self { client }
+    }
+}
```

Comment:
https://docs.rs/sqlx-core/0.8.6/sqlx_core/pool/struct.Pool.html

> `Pool` is Send, Sync and Clone. It is intended to be created once at the start of your application/daemon/web server/etc. and then shared with all tasks throughout the process’ lifetime.

Looks like `Postgres` would benefit from this approach as well. Do you think it would make sense to update the `Postgres` builder as well in this PR?